collections/_article/dev-snapshot-godot-4-5-dev-4.md

```diff
@@ -0,0 +1,125 @@
+
+## Highlights
+
+In case you missed them, see the [4.5 dev 1](/article/dev-snapshot-godot-4-5-dev-1/), [4.5 dev 2](/article/dev-snapshot-godot-4-5-dev-2/), and [4.5 dev 3](/article/dev-snapshot-godot-4-5-dev-2/) release notes for an overview of some key features which were already in that snapshot, and are therefore still available for testing in dev 4.
+
+### Move 3D physics interpolation to `SceneTree`
+
+The old design of the 3D interpolation system was fundamentally flawed, as it operated under the assumption that the scene side wouldn't require access to interpolated xforms. This isn't something that could've just been "patched in" either, thanks to the multithreaded design, command queue, and stalling. Lacking any sort of workaround, [lawnjelly](https://github.com/lawnjelly) has taken a break from being the person responsible for 90% of 3.x code nowadays & [forward-ported](https://github.com/godotengine/godot/pull/103685) a solution via [GH-104269](https://github.com/godotengine/godot/pull/104269). This addresses the problem by porting all logic to the `SceneTree`, but while completely retaining the existing API!
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/dev-snapshot-godot-4-5-dev-5/3d-fti-scenetree.webm?1" type="video/webm">
+</video>
+
+### Export variables as `Variant`
+
+Despite both arrays and dictionaries technically supporting `Variant` values, this functionality was never actually exposed in isolation. That is: it was impossible to export a variable of type `Variant` directly. [Tomasz Chabora](https://github.com/KoBeWi) found this limitation quite silly, so they've taken to addressing this grave injustice with [GH-89324](https://github.com/godotengine/godot/pull/89324). Now users are granted extra flexibility with their exports, as the option is now available to change not only the variable, but the type itself.
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/dev-snapshot-godot-4-5-dev-5/export-variant.webm?1" type="video/webm">
+</video>
+
+### Stackable outlines on `Label`
+
+Have you ever been in the situation where you want to add fancy outline/shadow effects to your text, only to realize that you're stuck with a single iteration of each? Sure, you can double-up the amount of text objects directly atop one-another, but that's cumbersome & doesn't account for outlines affecting other outlines. There must be a better way! Well, thanks to the efforts of [Delsin-Yu](https://github.com/Delsin-Yu), users no longer have to struggle with this moral conundrum. Instead, they can simply take advantage of [GH-104731](https://github.com/godotengine/godot/pull/104731) adding support for stacked layers of effects; no cumbersome workarounds required.
```

Comment:
```suggestion
Have you ever been in the situation where you want to add fancy outline or shadow effects to your text, only to realize that you're stuck with only one of each? Sure, you can double-up the amount of text directly atop one-another, but that's cumbersome and doesn't account for outlines affecting other outlines. There must be a better way! Well, thanks to the efforts of [Delsin-Yu](https://github.com/Delsin-Yu), users no longer have to struggle with this moral conundrum. Instead, they can simply take advantage of [GH-104731](https://github.com/godotengine/godot/pull/104731) adding support for stacked layers of effects; no cumbersome workarounds required.
```
You went really hard in hyping this one, too 😅 I don't know how to feel about it

Reply:
I didn't know what to say that wasn't self-evident, so I wrote it like an infomercial instead